src/api/index.tsx

```diff
@@ -184,17 +178,36 @@ export const callApi = async <
     body = JSON.stringify(bodyData);
   }
 
-  type QueryWithLang = {
-    [key: string]: string;
-  };
+  if (endpointUrl && endpoint) {
```

Comment:
I feel like we've looked at a similar thing before, but isn't there a way to define this with TypeScript?

Reply:
I thought we tried before but abandoned it because it required a [discriminated union](https://basarat.gitbooks.io/typescript/docs/types/discriminated-unions.html) which was too complicated? We could maybe do it with [function overloading](https://howtodoinjava.com/typescript/function-overloading/) but our code style enforces anonymous functions which don't support that.

I dunno, adding an if statement and one test seems like a good way to solve it.

Reply:
Well, huh, I can't find the discussion we had on this. If you had come up with a typed solution for this that I'm forgetting about, let me know.

Reply:
I don't have the exact syntax handy, and I think you're right that maybe we looked at using a discriminated union, but I'm not sure why we'd need one. The basic idea is this: create two types, each of which look like `CallApiParams` except that one (which we'll call `TypeA`) only has an `endpoint` property and the other one (which we'll call `TypeB`) only has an `endpointUrl` property. Then in place of `CallApiParams` we use `TypeA | TypeB`. Will that not work?